Sources/MongoSwift/MongoClient.swift

```diff
@@ -365,6 +404,7 @@ public class MongoClient {
         self.decoder = BSONDecoder(options: options)
         self.sdamEventHandlers = []
         self.commandEventHandlers = []
+        // commandEvents and sdamEvents are initialized when called
```

Comment:
Subject to change because of the potential change in approach with the on-demand, but the comment was left to show that 2 of the fields were initialized in that un-intuitive manner instead of in the `init` method. From what I gleaned, I thought that our method wouldn't initialize anything until `client.commandEvents` is called, which is what the comment is referring to. Altered comment for now to reflect the handler initialization, but again subject to change based on future approach

Reply:
oh I see. I read this as "commandEvents and sdamEvents are initialized when `initializeMonitoring()` is called", but you meant "commandEvents and sdamEvents are initialized when they are first accessed by the user".
but, yeah given the conversation above and proposed switch to not actually storing the single streams on the client I think this will be irrelevant anyway